test/junit/scala/collection/immutable/LazyListTest.scala

```diff
@@ -14,6 +14,34 @@ import scala.util.Try
 @RunWith(classOf[JUnit4])
 class LazyListTest {
 
+  @Test
+  def serialization(): Unit = {
+    import java.io._
+
+    def serialize(obj: AnyRef): Array[Byte] = {
+      val buffer = new ByteArrayOutputStream
+      val out = new ObjectOutputStream(buffer)
+      out.writeObject(obj)
+      buffer.toByteArray
+    }
+    def deserialize(a: Array[Byte]): AnyRef = {
+      val in = new ObjectInputStream(new ByteArrayInputStream(a))
+      in.readObject
+    }
+
+    def serializeDeserialize[T <: AnyRef](obj: T) = deserialize(serialize(obj)).asInstanceOf[T]
+
+    val l = LazyList.from(10)
+    val ld = serializeDeserialize(l)
+    assertEquals(ld.toString, "LazyList(<not computed>)")
```

Comment:
this doesn't really test anything, as the `LazyList` created from an `ArrayBuffer` will still not have computed the first element, even though it's obviously computed by the `ArrayBuffer`. we need some way of having each element evaluation be side effecting, such as adding to a counter. and then we can check the counter after deserializing? I don't know if that actually works

Reply:
no, I mean, this won't actually tell you whether or not the part that was unevaluated remained unevaluated

Reply:
OK, I see. The test is not very useful, I'll trim it down.